lib/money/currency.rb

```diff
@@ -34,18 +34,17 @@ def initialize(method, currency, attribute)
     class UnknownCurrency < ArgumentError; end
 
     class << self
-      alias_method :original_new, :new
       def new(id)
         id = id.to_s.downcase
         unless stringified_keys.include?(id)
           raise UnknownCurrency, "Unknown currency '#{id}'"
         end
 
-        @@mutex.synchronize { instances[id] } || super
+        @@mutex.synchronize { instances[id] ||= super }
```

Comment:
I'm afraid it won't work, because access to `instances[id]` should be synchronized as well, otherwise it causes random failures.